### PR TITLE
Deprecate functions that are planned removed in v2.0.0

### LINKED ISSUE #191: Consider removing functionality for fmu-ensemble 2.0
* `to_disk()` and related code is perhaps used by zero people. If so, it is tempting not to drag it along.
* fmu.ensemble.etc is deprecated and should be removed
* unrst/init  access function in ensemble/realization should be removed. The way to this in the future would be go  through ecl2df.
* `cache_eclsum` should maybe go away, it   is not meaningful when run concurrently. The  batch feature will mitigate the problem.
* The `stacked` option to `load_smry()` (and others?) should be deleted (raises NotImplementedError)
* `realization.get_smryvalue()` should not be used, is redundant.

### PR DESCRIPTION
**Pull request by berland**
See #191

### CODE REVIEW

**Comment by asnyv** [comment]
Use DeprecationWarning instead?

**Comment by berland** [comment]
> Use DeprecationWarning instead?

DeprecationWarnings are by default not visible to users, according to Python docs they are meant for developers. FutureWarning is by default visible to users.

**Comment by asnyv** [comment]
> > Use DeprecationWarning instead?
> 
> DeprecationWarnings are by default not visible to users, according to Python docs they are meant for developers. FutureWarning is by default visible to users.

Well, that was part of the point 😅

**Comment by asnyv** [comment]
fmu-ensemble tends to write many warnings considering the iterative nature of it 😬

**Comment by berland** [comment]
> fmu-ensemble tends to write many warnings considering the iterative nature of it

Yes,  this patch has not been tried live. It might emit one warning pr. realization, but if that causes user action, it is the intention..

File: src/fmu/ensemble/ensembleset.py [hunk 1]
```diff
@@ -664,6 +674,18 @@ def get_smry(
...
+        if cache_eclsum is not None:
+            warnings.warn(
+                (
+                    "cache_eclsum option to get_smry() is deprecated and "
+                    "will be removed in fmu-ensemble v2.0.0"
+                ),
+                FutureWarning,
+            )
+        else:
+            cache_eclsum = False
```
Comment (asnyv) [inline src/fmu/ensemble/ensembleset.py:687]:
```suggestion
```

File: src/fmu/ensemble/ensembleset.py [hunk 2]
```diff
@@ -744,6 +775,10 @@ def get_wellnames(self, well_match=None):
             summary file or no matched well names.
 
         """
+        warnings.warn(
+            "fmu.ensemble.etc is deprecated and will be removed in later versions.",
+            FutureWarning,
+        )
```
Comment (asnyv) [inline src/fmu/ensemble/ensembleset.py:781]: Not sure if it is supposed to be deprecated or not, but the message is at least wrong.
```suggestion
        warnings.warn(
            "ensembles.get_wellnames() is deprecated and will be removed in later versions.",
            FutureWarning,
        )
```

File: src/fmu/ensemble/realization.py [hunk 3]
```diff
@@ -1192,6 +1216,14 @@ def get_smryvalues(self, props_wildcard=None):
             a dataframe with values. Raw times from UNSMRY.
             Empty dataframe if no summary file data available
         """
+        warnings.warn(
+            (
+                "realization.get_smryvalues() is deprecated and "
+                "will be removed in fmu-ensemble v2.0.0"
```
Comment (asnyv) [inline src/fmu/ensemble/realization.py:1222]: Add suggestion for alternative? Probably not in much use anyways.
```suggestion
                "realization.get_smryvalues() is deprecated and "
                "will be removed in fmu-ensemble v2.0.0."
                "Use realization.get_smry() instead."
```

File: src/fmu/ensemble/realization.py [hunk 4]
```diff
@@ -1079,6 +1091,18 @@ def get_smry(
         Returns empty dataframe if there is no summary file, or if the
         column_keys are not existing.
         """
+
+        if cache_eclsum is not None:
+            warnings.warn(
+                (
+                    "cache_eclsum option to load_smry() is deprecated and "
```
Comment (asnyv) [inline src/fmu/ensemble/realization.py:1098]:
```suggestion
                    "cache_eclsum option to get_smry() is deprecated and "
```

**Review by asnyv — CHANGES_REQUESTED** [review]

File: src/fmu/ensemble/ensembleset.py (hunk 1 above)
Comment (berland) [reply src/fmu/ensemble/ensembleset.py:687]: fixed

File: src/fmu/ensemble/ensembleset.py (hunk 2 above)
Comment (berland) [reply src/fmu/ensemble/ensembleset.py:781]: fixed

File: src/fmu/ensemble/realization.py (hunk 4 above)
Comment (berland) [reply src/fmu/ensemble/realization.py:1098]: thanks!

File: src/fmu/ensemble/realization.py (hunk 3 above)
Comment (berland) [reply src/fmu/ensemble/realization.py:1222]: fixed

**Comment by asnyv** [comment]
LGTM 👍

**Review by asnyv — APPROVED** [review]

**Pull request merged**